File: game/pendu.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

void displayCurrentState(char *word, char *guessedword)
{
    printf("Current word: %s\n", guessedword);
}

int iswordGuessed(char *word, char *guessedword) {
    return strcmp(word, guessedword) == 0;
}

void updateGuessedword(char *word, char *guessedword, char guess)
{
    for (int i = 0; i < strlen(word); i++) {
        if (tolower(word[i]) == guess) {
            guessedword[i] = word[i];
        }
    }
}

void getGuess(char *guess)
{
    printf("Enter a letter: ");
    scanf(" %c", guess);
    *guess = tolower(*guess);
}

int main(void)
{
    char word[] = "programming";
    int wordLength = strlen(word);
    char guessedword[wordLength + 1];
    memset(guessedword, '_', wordLength);
    guessedword[wordLength] = '\0';
    int attempts = 6;
    char guess;

    while (attempts > 0 && !iswordGuessed(word, guessedword)) {
        displayCurrentState(word, guessedword);
        getGuess(&guess);

        if (strchr(word, guess)) {
            updateGuessedword(word, guessedword, guess);
        } else {
            attempts--;
            printf("Wrong guess! Attempts left: %d\n", attempts);
        }
    }
    if (iswordGuessed(word, guessedword)) {
        printf("Congratulations! You've guessed the word: %s\n", word);
    } else {
        printf("You've run out of attempts! The word was: %s\n", word);
    }
    return 0;
}

File: game/pendu.py
def display_current_state(word, guessed_word):
    print("Current word:", " ".join(guessed_word))

def is_word_guessed(word, guessed_word):
    return word == "".join(guessed_word)

def update_guessed_word(word, guessed_word, guess):
    for i, letter in enumerate(word):
        if letter == guess:
            guessed_word[i] = guess

def get_guess():
    guess = input("Enter a letter: ").lower()
    return guess

def main():
    word = "programming"
    guessed_word = ['_' for _ in word]
    attempts = 6
    
    while attempts > 0 and not is_word_guessed(word, guessed_word):
        display_current_state(word, guessed_word)
        guess = get_guess()

        if guess in word:
            update_guessed_word(word, guessed_word, guess)
        else:
            attempts -= 1
            print(f"Wrong guess! Attempts left: {attempts}")

    if is_word_guessed(word, guessed_word):
        print(f"Congratulations! You've guessed the word: {word}")
    else:
        print(f"You've run out of attempts! The word was: {word}")

if __name__ == "__main__":
    main()
